### PR TITLE
Add Sync e2e test flows

### PR DESCRIPTION
**Pull request by loremattei**
Task/Issue URL: https://app.asana.com/0/0/1205914590479611/f
Tech Design URL:
CC:

**Description**:

This PR adds the critical e2e tests identified in the task above. 

Most of them will need an update soon as the new sync setup flow is implemented, but the most interesting part of this PR is that it sets up the integration with the JS test script, so it's still worth it to merge it.

**Steps to test this PR**:
1. Verify CI is green.

To test locally:
2. Download https://github.com/duckduckgo/sync_crypto (You want to use [this](loremattei/make-run-locally) branch if the PR has not been merged yet).
3. Run `node index.js` and copy the recovery code that the script outputs.
4. Open an iOS simulator and install the iOS app. 
5. Run `maestro test -e CODE=<Paste the code here> ./.maestro/sync_tests`.
6. When finished, delete the test sync account (The easiest way to do it is using the dedicated option in the iOS app).

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .maestro/sync_tests/03_recover_account.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- clearState
+- launchApp
+- runFlow: 
+    when: 
+      visible: 
+        text: "Let’s Do It!"
+        index: 0
+    file: ../shared/onboarding.yaml
+
+# <WORDAROUND>This is a workaround to:
```
Comment (ayoy) [inline .maestro/sync_tests/03_recover_account.yaml]: typo in WORKAROUND (and in the closing tag below :) and also in other occurrences in other files)

File: .github/workflows/sync-end-to-end.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: Sync-End-to-End tests
+
+on:
+  push
```
Comment (ayoy) [inline .github/workflows/sync-end-to-end.yml]: This, as I understand, will be removed before merging and we'll keep it running on schedule (as per the code below that's currently commented out?)

**Review by ayoy — APPROVED** [review]
LGTM! I couldn't test locally because I don't have Xcode 14.3, but I went through tests code and it makes perfect sense, and also the CI is green so I think it's good to go. Thanks for handling this @loremattei!

File: .github/workflows/sync-end-to-end.yml (hunk 2 above)
Comment (loremattei) [reply .github/workflows/sync-end-to-end.yml]: Oh, yes! Sorry I didn't mention in the description 🤦‍♂️

Comment (loremattei) [reply .github/workflows/sync-end-to-end.yml]: Updated now!

**Pull request merged**